File: Demo/JXC.WPF/Common/Commands/ShowBill.cs
/*******************************************************
 * 
 * 作者：胡庆访
 * 创建时间：20120418
 * 说明：此文件只包含一个类，具体内容见类型注释。
 * 运行环境：.NET 4.0
 * 版本号：1.0.0
 * 
 * 历史记录：
 * 创建文件 胡庆访 20120418
 * 
*******************************************************/

using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Security.Permissions;
using System.Text;
using Rafy;
using Rafy.WPF.Command;
using Rafy.MetaModel.Attributes;
using Rafy.MetaModel.View;
using Rafy.WPF;
using JXC.WPF.Templates;
using Rafy.WPF.Controls;

namespace JXC.Commands
{
    [Command(Label = "查看", GroupType = CommandGroupType.View)]
    public class ShowBill : ListViewCommand
    {
        public ShowBill()
        {
            this.Template = new ReadonlyBillTemplate();
        }

        protected UITemplate Template;

        public override bool CanExecute(ListLogicalView view)
        {
            return view.Current != null;
        }

        public override void Execute(ListLogicalView view)
        {
            //弹出窗体显示详细面板
            this.Template.EntityType = view.EntityType;
            var ui = this.Template.CreateUI();

            var detailView = ui.MainView.CastTo<DetailLogicalView>();
            detailView.Data = view.Current;

            App.Windows.ShowDialog(ui.Control, w =>
            {
                w.Buttons = ViewDialogButtons.Close;
                w.Title = this.Meta.Label.Translate() + " " + view.Meta.Label.Translate();
            });
        }
    }
}

File: trunk/Demo/JXC.Web/Common/Commands/ShowBill.js
/*******************************************************
 * 
 * 作者：胡庆访
 * 创建时间：201201
 * 说明：
 * 运行环境：.NET 4.0
 * 版本号：1.0.0
 * 
 * 历史记录：
 * 创建文件 胡庆访 201201
 * 
*******************************************************/

Rafy.defineCommand('Jxc.ShowBill', {
    meta: { text: "查看", group: "view" },

    _template: '',

    constructor: function () {
        this.callParent(arguments);

        this._template = new Jxc.ReadonlyBillTemplate();
    },

    canExecute: function (listView) {
        return listView.getCurrent() != null;
    },

    execute: function (listView, source) {
        var ui = this._template.createUI(listView.getModel());
        var view = ui.getView();
        var item = listView.getCurrent();
        view.setCurrent(item);

        var win = Rafy.Window.show({
            title: '查看' + listView.getMeta().label,
            items: ui.getControl(), animateTarget: source
        });
    },
    getTemplate: function () { return this._template; },
    setTemplate: function (value) { this._template = value; }
});